### PR TITLE
add config to use mac os tts instead of elevenlabs or gtts

### PR DESCRIPTION
**Pull request by Wladastic**
As a Mac OS user, I prefer the builtin TTS of Siri etc.

I want to be able to configure that the builtin TTS of Mac OS is used instead of Google's tts.

As I am a lazy dude, here is chatgpt explaining how to change the voice:

**To change the default voice for spoken text in macOS, follow these steps:**
1. Open System Preferences by clicking on the Apple menu in the top-left corner of your screen and selecting "System Preferences."
2. Click on "Accessibility."
3. In the left sidebar, click on "Speech" (in macOS 10.15 or earlier) or "Spoken Content" (in macOS 11 or later).
4. Locate the "System Voice" dropdown menu (or "Voice" dropdown menu in macOS 11 or later) and click on it to reveal the list of available voices.
5. Select the voice you want to set as the default. You can click on the "Play" button next to the dropdown menu to preview the selected voice.
6. Close the "Accessibility" window to save your changes. The selected voice is now the default voice for spoken text in macOS.

### CODE REVIEW

File: scripts/speak.py [hunk 1]
```diff
@@ -3,11 +3,12 @@
...
 cfg = Config()
-import gtts
 
+# Remove the import of gtts
+# import gtts
 
-# TODO: Nicer names for these ids
-voices = ["ErXwobaYiN019PkySvjV", "EXAVITQu4vr4xnSDxMaL"]
+# Change voices to macOS voice identifiers
+voices = ["com.apple.speech.synthesis.voice.siri_female", "com.apple.speech.synthesis.voice.siri_male"]
```
Comment (Torantulino) [inline scripts/speak.py]: This list of voices is for elevenlabs.io, not GTTS.

**Comment by Torantulino** [comment]
Unless I'm missing something, I'd imagine switching to MacOS text to speech would cut out non-mac users?

Can you explain why you've submitted this PR?

File: scripts/speak.py (hunk 1 above)
Comment (nponeccop) [reply scripts/speak.py]: He also removed Windows support by hardcoding to macos below and not making it configurable.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Ensure it still works on Windows without extra configuration. So, make it configurable.

File: scripts/speak.py (hunk 1 above)
Comment (Wladastic) [reply scripts/speak.py]: My bad, I didn't mean to cut out windows users. 
My bot seems to have only pushed the first commit. 
will fix this in a few hours

**Comment by Wladastic** [comment]
I will push the remaining changes shortly.

File: scripts/speak.py [hunk 2]
```diff
@@ -5,7 +5,6 @@
 cfg = Config()
 import gtts
 
-
```
Comment (nponeccop) [inline scripts/speak.py]: An unrelated whitespace change creeped in.

File: scripts/speak.py [hunk 3]
```diff
@@ -38,11 +37,16 @@ def gtts_speech(text):
...
-        gtts_speech(text)
+        if cfg.use_mac_os_tts == 'True':
+            macos_tts_speech(text)
+        else:
+            gtts_speech(text)
     else:
         success = eleven_labs_speech(text, voice_index)
         if not success:
             gtts_speech(text)
-
```
Comment (nponeccop) [inline scripts/speak.py]: Another one

File: .env.template [hunk 4]
```diff
@@ -11,4 +11,5 @@ OPENAI_API_BASE=your-base-url-for-azure
 OPENAI_API_VERSION=api-version-for-azure
 OPENAI_DEPLOYMENT_ID=deployment-id-for-azure
 IMAGE_PROVIDER=dalle
-HUGGINGFACE_API_TOKEN=
\ No newline at end of file
+HUGGINGFACE_API_TOKEN=
+USE_MAC_OS_TTS=False
```
Comment (nponeccop) [inline .env.template:15]: I think we don't need an empty line here

**Review by nponeccop — CHANGES_REQUESTED** [review]
Good job! Now it's eye candy. The only remaining problems are 3 added empty lines.

File: .env.template (hunk 4 above)
Comment (Wladastic) [reply .env.template:15]: it just looks like it, there is no line between 14 and 15

Comment (nponeccop) [reply .env.template:15]: LGTM then

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@Wladastic Conflicts again. This is good news; we did a massive merge of minor PRs. Resolve.

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@Torantulino at least you can remove the labels now 🥇

**Comment by Torantulino** [comment]
Thanks for fixing this up, and thank you @nponeccop for your excellent help.
